### ISSUE TITLE
Consider restructuring the project modules (lib/gui/plugins)

### ISSUE THREAD

**Issue by toxeus** [spesmilo/electrum#4279]
This is a minor issue but still worth reporting.

`test_electrum_seed_2fa` was added in this [commit](https://github.com/spesmilo/electrum/commit/6d88eab005c306eb9fda6249cf387656c56a2122) which imports `plugins.trustedcoin` which imports [`electrum` and its submodules](https://github.com/spesmilo/electrum/blob/f6cb26be1fa1d1b3de92e5fce6c24f60d70d357a/plugins/trustedcoin/trustedcoin.py#L33-L44). As a consequence breaking changes in `electrum` or its submodules will not break the test until the `electrum` package is reinstalled. Even worse, a `git pull` might suddenly result in the test failing, assuming that the test and trustedcoin were changed.

**Comment by toxeus** [comment]
@SomberNight in my personal projects I use the following project structure which isn't affected at all by the problems electrum currently faces.

```sh
repo
├── script
├── pkg1
├── pkg2
└── ...
```

The [hacks in the main script](https://github.com/spesmilo/electrum/blob/376a815458d7765c3e3b4c22b74552bd4702b163/electrum#L68-L72) are not needed and the unit tests use the newest code from the repo because Python puts [the directory containing the input script](https://docs.python.org/3.6/tutorial/modules.html#the-module-search-path) on top of its module search path (consequently `pytest` needs to be executed from the projects root dir; not sure if this is already the case).

For the electrum project the structure would therefore look like the following (only listing relevant files and folders).

```sh
electrum
├── electrum.py
├── electrum
├── electrum_gui
└── electrum_plugins
```

**Comment by SomberNight** [comment]
@ecdsa @ysangkok @bauerj 
probably of interest to all of you

**Comment by SomberNight** [comment]
(context: https://github.com/spesmilo/electrum/pull/4286#issuecomment-383161468 )

**Comment by ysangkok** [comment]
@toxeus Try this: https://github.com/ysangkok/electrum/tree/file_layout_change

What do you think? I haven't used your suggested layout since it requires more hacks when there is no module that contains all relevant sources. This still needs to be tested on all platforms except Linux.

**Comment by toxeus** [comment]
@ysangkok sorry for the late reply. I don't know how to review 265 commits efficiently.

![screenshot_2018-07-07 ysangkok electrum 1](https://user-images.githubusercontent.com/7702854/42409168-96c1c2fc-81d6-11e8-8b65-c1831d231916.png)

This branch seems to have different kinds of changes intermingled.

![screenshot_2018-07-07 ysangkok electrum](https://user-images.githubusercontent.com/7702854/42409167-96a07cb4-81d6-11e8-91fe-87207b97b737.png)

**Comment by ysangkok** [comment]
It's just one commit on top of a now-rebased lightning branch. But after reading your other comment, I don't see any particular problems with your approach.

**Comment by toxeus** [comment]
> It's just one commit on top of a now-rebased lightning branch.

I see. Sorry for the misunderstanding then :sweat_smile:

**Comment by ysangkok** [comment]
Fixed in 097ac144d976eb46dff809e1809783dc78ab6d8b . Thanks for the report.
